src/server/modes/charts/plugins/datalens/preparers/bar-x/d3.ts

```diff
@@ -23,14 +23,17 @@ type OldBarXDataItem = {
 } | null;
 
 export function prepareD3BarX(args: PrepareFunctionArgs): ChartKitWidgetData {
-    const {shared, labels, placeholders} = args;
+    const {shared, labels, placeholders, applyDefaultSorting = true} = args;
```

Comment:
Below, the default sorting is applied if there are no fields in the corresponding placeholder. It seems to me that in your case it would be more appropriate to use a flag to disable sorting. Something like `disableSorting` maybe?